Modules/database_manager.py

```diff
@@ -0,0 +1,386 @@
+        """
+
+        Args:
+            constraint_column_name: names of the unique columns.
+                These MUST be set as unique at table-creation
+            no_conflict_resolution: If there are no unique columns in your table, set this to True
+            table_name: name of table to insert value into
+            values: tuple with values matching the rows to insert into
+            skip_double_dash_test: skip the crude SQLInjection test if it breaks your request,
+                    implement your OWN CHECK!
+
+        Returns: None
+
+        """
+
+    @abstractmethod
+    async def update_row(self, table_name: str, connector: str,
+                         values: Dict[str, Union[str, datetime.datetime]],
+                         condition: Dict[str, str] = None,
+                         skip_double_dash_test=False) -> None:
+        """
+
+         Args:
+             table_name: name of table to update
+             connector: Connector used to connect conditions. Must be suported by the DB
+             values: tuple with values matching the rows to insert into
+             condition: conditions, connected by "equals"
+             skip_double_dash_test: skip the crude SQLInjection test if it breaks your request,
+                     implement your OWN CHECK!
+
+         Returns: None
+
+         """
+
+    @abstractmethod
+    async def delete_row(self, table_name: str, connector: str, condition: Dict[str, str] = None,
+                         skip_double_dash_test: bool = False) -> None:
+        """
+
+        Args:
+            table_name: name of table to delete row form
+            connector: Connector used to connect conditions. Must be suported by the DB
+            condition: conditions, connected by "equals"
+            skip_double_dash_test: skip the crude SQLInjection test if it breaks your request,
+                    implement your OWN CHECK!
+
+        Returns: None
+
+        """
+
+
+# noinspection SqlNoDataSourceInspection
+class DatabaseManager(DBMApi, Singleton):
+    _enabled = True
+
+    # for testing only, as a makeshift "session" scope
+    marker = None
+
+    cursor = None
+
+    @property
+    def enabled(self):
+        return self._enabled
+
+    @enabled.setter
+    def enabled(self, value: bool):
+        if not isinstance(value, bool):
+            raise ValueError(f"{type(value)} is not a valid type for setting a bool!")
+        if value == self._enabled:
+            # no need to change it if no change is required
+            return
+        if value:
+            try:
+                self.init_connection()
+                self._enabled = True
+            except pyodbc.Error as ex:
+                log.error(f"Connection reinitialization failed! {ex}")
+                raise
+        else:
+            self.connection.close()
+            self._enabled = False
+
+    def _execute(self, *args, **kwargs):

```

Comment:
We can simplify this enabled check.

```py
if not self.enabled:
            log.warning("Tried to execute a DB call with the DB access being disabled.")
            raise RuntimeError("Check if the DB-module is enabled before executing code!"
                               " Currently it is NOT!")

# contents of else on line 169

This would save one layer of complexity